tests/buffer/buffer_constructors_common.h

```diff
@@ -200,278 +198,266 @@ class buffer_ctors {
       }
       std::vector<T> ref(cont1.size());
       std::iota(ref.begin(), ref.end(), 0);
-      if (!std::equal(ref.cbegin(), ref.cend(), cont1.cbegin(),
-                      value_operations::are_equal<T, T>) ||
-          !std::equal(ref.cbegin(), ref.cend(), cont2.cbegin(),
-                      value_operations::are_equal<T, T>) ||
-          !alloc_log.str().empty())
-        FAIL(log, "(Container, allocator) constructor fail.");
+      CHECK(std::equal(ref.cbegin(), ref.cend(), cont1.cbegin(),
+                       value_operations::are_equal<T, T>));
+      CHECK(std::equal(ref.cbegin(), ref.cend(), cont2.cbegin(),
+                       value_operations::are_equal<T, T>));
+      CHECK(alloc_log.str().empty());
     }
 
-    /* check (shared pointer, range) constructor*/
     {
-      std::shared_ptr<T> data(new T[size]);
+      INFO("check (shared pointer, range) constructor");
+      std::shared_ptr<T> data(new T[size], deleter<T>());
```

Comment:
By looking at https://en.cppreference.com/w/cpp/memory/shared_ptr/shared_ptr I do not think you need to provide a specific deleter, even in old C++17. It should use the right deleter by default.
This would be necessary for `std::make_shared`, which has been fixed only in C++20.